### DISCUSSION TITLE
Nextjs in docker always reload when routing another menu

### DISCUSSION THREAD

**Discussion by hnhai** [vercel/next.js#40262] [Help] [answered]
- Local: yarn install -> yarn build -> yarn start -> visit localhost:3000. Click menu tabs -> Don't reload page.
- Docker: yarn install -> yarn build -> copy resource to container -> yarn start & expose port 3000 to host -> Visit localhost:3000. Click menu tabs -> Reload every time click on a menu tab.
I wan't keep SPA on docker. Thank you.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Do you have a reproduction demo or repository? This shouldn't happen at all. There must be something else going on that's triggering page reloads.

How are you using the next/link? It is really hard to tell without looking at some code. I set up a small example and couldn't reproduce what you are seeing.

**Reply by hnhai** [reply]
Sorry @icyJoseph this is private app of my company, i can't share you. I will create a clone app later. 
Can you share your example ?

**Reply by icyJoseph** [reply]
I mean, it is just two pages linking to each other:

```tsx
import type { NextPage } from "next";
import Link from "next/link";

const Home: NextPage = () => {
  return (
    <div>
      Home <Link href="/blog">blog</Link>
      <div>more stuff</div>
    </div>
  );
};

export default Home;
```

And blog links back to home, then I used the docker config from https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile, and it runs swell...

**Reply by hnhai** [reply]
I'm using Nextjs version 8, don't have standalone configaration.
I just copy public, .next and node_modules folder in to container.
I will try upgrade Nextjs to newest version, thank for your support

**Reply by icyJoseph** [reply]
Aha, version 8... I do remember those days... still strange that it has this behaviour.  Not really sure what could the problem be. Finding docs for version 8 is not very easy these days.

**Reply by hnhai** [reply]
Yes, I not sure too. I will try upgrade to lastest version using new Dockerfile and response here.
Thanks for your support

**Reply by hnhai** [reply]
@icyJoseph 
I add lib: babel-plugin-import and it works fine.
